src/task/op.rs

```diff
@@ -64,21 +64,24 @@ impl OpRuntime {
         let user_data = self.ops.insert(RegisteredOp::new(Some(RawOp::new(op))));
         let op = unsafe { self.ops.get_unchecked_mut(user_data) };
         // Safety: `user_data` corresponds to `op` inserted which has type `T`.
-        (unsafe { Key::new(user_data) }, op.op.as_mut().unwrap())
+        (
+            unsafe { Key::new(user_data as u64) },
+            op.op.as_mut().unwrap(),
+        )
     }
 
     pub fn insert_dummy(&mut self) -> Key<()> {
-        Key::new_dummy(self.ops.insert(RegisteredOp::new(None)))
+        Key::new_dummy(self.ops.insert(RegisteredOp::new(None)) as u64)
     }
 
     pub fn update_waker<T>(&mut self, key: Key<T>, waker: Waker) {
-        if let Some(op) = self.ops.get_mut(*key) {
+        if let Some(op) = self.ops.get_mut(usize::try_from(*key).expect("in range")) {
```

Comment:
You see, you changed the type of `user_data` to u64. It's right for io-uring, but here is more complex. Nowadays even the users of 32-bit Windows is decreasing, let alone the users of 32-bit Linux. I cannot imagine a user with 32-bit Linux and io-uring. I don't like this change.

Reply:
I want to pack structure into `user_data` to dispatch callbacks between multiple IO clients with a dedicated queue for each IO operation (known max IO depth). u64 gives fixed size.

So in general `user_data` could be used not as a index in global operations queue.

Reply:
`user_data` is used a key when cancelling it (see `AsyncCancel` opcode). In `compio`, it is also used for `Token` in `mio`, and a hashmap index for IOCP. I'm here considering other platforms for you to make sure the public, cross-platform general interface is suitable for all different kind of drivers.